server/webapp/WEB-INF/rails.new/app/presenters/api_v1/config/artifact_representer.rb

```diff
@@ -39,17 +39,21 @@ def type
       end
 
       def errors
-        mapped_errors = artifact.errors
-
-        if src_errors = mapped_errors.delete('src')
-          mapped_errors['source'] = src_errors
+        mapped_errors = {}
+        artifact.errors.each do |key, value|
+          mapped_errors[matching_error_key(key)] = value
         end
+        mapped_errors
+      end
 
-        if dest_errors = mapped_errors.delete('dest')
-          mapped_errors['destination'] = dest_errors
-        end
+      private
+      def error_keys
+        {"src" => "source", "dest" => "destination"}
+      end
 
-        mapped_errors
+      def matching_error_key key
+        return error_keys[key] if error_keys[key]
+        key
```

Comment:
`error_keys[key] || key` reads better :)

I like this pattern. Can we extract this into a class with usage that looks like this, and use it everywhere?

```ruby
def errors
  ErrorMapper.new(artifact.errors, 'src' => 'source', 'dest' => 'destination').mapped_errors
end
```